kmk/extensions/lock_status.py

```diff
@@ -15,11 +15,12 @@ class LockCode:
 
 class LockStatus(Extension):
     def __init__(self):
-        self.report = 0x00
+        self.report = None
```

Comment:
Why is necessary to set the report to `None`? It's always a byte and the "no locks are enabled" state is `0x00`.

Reply:
There is no way to know what the state it at the initialization of this class. Therefore inferring that the state is `0x00` is incorrect and can lead to false information being reported in people's code.

Reply:
The only reason that this wouldn't update once the keyboard firmware is running is when you poll the state once during the setup and never again. And at this point `None` would throw an exception.
I don't see how initializing it to `None` would help.
Sorry if I'm being tedious, I just don't want to "fix" something that's not broken.

Reply:
I made the change based on my findings when testing my original change. Now that I think about it though, my callback was inside of a conditional that only called the function if the report changes. Since report was initializing as 0 and the last report was 0, there was no change and my function was never getting called.

I could see someone wanting to do the same thing so that they could only trigger code when the report actually changes.

As far as exceptions go, aren't all exceptions caught and ignored unless debugging in which case it just prints to log? It isn't good practice to purposely throw exceptions for no reason, but setting the value to a random value also doesn't sit well with me either. Technically, initializing it to None is the most a appropriate value at the time. I am open to suggestions here.

Reply:
> I could see someone wanting to do the same thing so that they could only trigger code when the report actually changes.

A `has_changed` class attribute for users to check against sounds reasonable.

> As far as exceptions go, aren't all exceptions caught and ignored unless debugging in which case it just prints to log? It isn't good practice to purposely throw exceptions for no reason, but setting the value to a random value also doesn't sit well with me either. Technically, initializing it to None is the most a appropriate value at the time. I am open to suggestions here.

Well, 0 is at least a valid value, and it meaning "no lock is active" isn't totally arbitrary as a fallback if in doubt. That value is only maintained for the first run of the firmware main loop anyway.
The reason that there's a check for `self.hid` and the report isn't polled at init, if I remember correctly, is that in certain circumstances the hid connectivity isn't there at all (split keyboard), delayed after init (at boot, BLE), or intermittently not availabel (due to BIOS/OS handover, or OS driver re-initializations).
Exceptions are only caught if any of the `get_xxx_lock` methods are called from within module / extension handlers, not during setup. And even then, something like a split not connected to USB may try to throw > 200 exceptions per second, which it won't manage, because it'll slow the entire device down to a crawl.

I think that could be somewhat mitigated by checking against the aforementioned future `has_changed` attribute. But as an internal API, the `get_xxx` methods should never be allowed to throw an exception in my opinion. Then I'd rather initialize it to 0x80 for example. That'll trigger the first `has_changed` without interfering with the `get_xxx` methods.

Reply:
Ah, yeah definitely use cases that I am not aware of.

Fun fact though, according to https://docs.circuitpython.org/en/latest/shared-bindings/usb_hid/index.html, `get_last_received_report` can return `None` so it isn't unprecedented.

Maybe a better idea would be to put appropriate checks in the code to avoid doing something that throws an exception.